org.envirocar.app/src/org/envirocar/app/views/logbook/LogbookActivity.java

```diff
@@ -214,6 +235,14 @@ public void onFuelingUploaded(Fueling fueling) {
      */
     private void downloadFuelings() {
         LOG.info("downloadFuelings()");
+
+        // If internet connection is ON , downloadFuel data else set background as no internet.
+        if (!isNetworkAvailable(getApplication())){
+            headerView.setVisibility(View.GONE);
+            newFuelingFab.setVisibility(View.GONE);
+            noInternetConnection();
+            showSnackbarInfo(R.string.error_not_connected_to_network);
+        }
```

Comment:
Use the `ContextInternetAccessProvider` class here as mentioned above. In addition, setting the subscription for the `FuelingObservable` should only be done, if internet is available.